test/MustSpendTotal.purs

```diff
@@ -0,0 +1,68 @@
+module Test.Ctl.MustSpendTotal
+  ( main
+  , suite
+  ) where
+
+import Prelude
+
+import Contract.Prim.ByteArray (byteArrayFromAscii)
+import Contract.Scripts (MintingPolicy)
+import Contract.TxConstraints
+  ( TxConstraints
+  , mustProduceAtLeast
+  , mustProduceAtLeastTotal
+  , mustSpendAtLeast
+  , mustSpendAtLeastTotal
+  )
+import Contract.Value (CurrencySymbol, TokenName)
+import Contract.Value as Value
+import Control.Monad.Error.Class (catchJust, liftMaybe)
+import Ctl.Examples.AlwaysMints (alwaysMintsPolicyMaybe)
+import Ctl.Internal.Test.TestPlanM (TestPlanM, interpret)
+import Data.Maybe (Maybe(Nothing))
+import Data.Tuple.Nested ((/\))
+import Effect (Effect)
+import Effect.Aff (Aff, launchAff_)
+import Effect.Class (liftEffect)
+import Effect.Exception (error)
+import Mote (group, test)
+import Test.Spec.Assertions (shouldEqual)
+
+-- Run with `spago test --main Test.Ctl.MustSpendTotal`
+main :: Effect Unit
+main = launchAff_ do
+  interpret suite
+
+suite :: TestPlanM (Aff Unit) Unit
+suite = do
+  group "mustXAtLeastTotal works correctly" do
+    test "mustSpendAtLeastTotal works correctly" $ do
+      testIsAuthomorhism mustSpendAtLeast mustSpendAtLeastTotal
+    test "mustProduceAtLeastTotal works correctly" $ do
+      testIsAuthomorhism mustProduceAtLeast mustProduceAtLeastTotal
+
+testIsAuthomorhism
+  :: forall i o
+   . (Value.Value -> TxConstraints i o)
+  -> (TxConstraints i o -> Value.Value)
+  -> Aff Unit
+testIsAuthomorhism valueToConstraint constraintToValue = do
+  token1 <- mkTokenName "Token 1"
+  token2 <- mkTokenName "Token 2"
+  cs <- mkCurrencySymbol alwaysMintsPolicyMaybe
+  let
+    value1 = Value.singleton cs token1 one
+    value2 = Value.singleton cs token2 one
```

Comment:
I think this must be a property test instead.

Reply:
Of course, you are right. I thought we have no property test on this project, but I checked and found I was wrong.